tests/api/v1/endpoints/test_privacy_request_endpoints.py

```diff
@@ -1610,3 +1613,173 @@ def test_resume_privacy_request(
                 "name": privacy_request.policy.name,
             },
         }
+
+
+class TestResumeWithManualInput:
+    @pytest.fixture(scope="function")
+    def url(self, db, privacy_request):
+        return V1_URL_PREFIX + PRIVACY_REQUEST_MANUAL_INPUT.format(
+            privacy_request_id=privacy_request.id
+        )
+
+    def test_manual_resume_not_authenticated(self, api_client, url):
+        response = api_client.post(url, headers={}, json={})
+        assert response.status_code == 401
+
+    def test_manual_resume_wrong_scope(self, api_client, url, generate_auth_header):
+        auth_header = generate_auth_header(scopes=[PRIVACY_REQUEST_READ])
+
+        response = api_client.post(url, headers=auth_header, json={})
+        assert response.status_code == 403
+
+    def test_manual_resume_privacy_request_not_paused(
+        self, api_client, url, generate_auth_header, privacy_request
+    ):
+        auth_header = generate_auth_header(scopes=[PRIVACY_REQUEST_CALLBACK_RESUME])
+        response = api_client.post(url, headers=auth_header, json=[{"mock": "row"}])
+        assert response.status_code == 400
+        assert (
+            response.json()["detail"]
+            == f"Invalid resume request: privacy request '{privacy_request.id}' status = in_processing."
+        )
+
+    def test_manual_resume_privacy_request_no_paused_location(
+        self, db, api_client, url, generate_auth_header, privacy_request
+    ):
+        auth_header = generate_auth_header(scopes=[PRIVACY_REQUEST_CALLBACK_RESUME])
+        privacy_request.status = PrivacyRequestStatus.paused
+        privacy_request.save(db)
+
+        response = api_client.post(url, headers=auth_header, json=[{"mock": "row"}])
+        assert response.status_code == 400
+        assert (
+            response.json()["detail"]
+            == f"Cannot resume privacy request '{privacy_request.id}'; no paused collection."
+        )
+
+    def test_resume_with_manual_input_collection_has_changed(
+        self, db, api_client, url, generate_auth_header, privacy_request
+    ):
+        """Fail if user has changed graph so that the paused node doesn't exist"""
+        auth_header = generate_auth_header(scopes=[PRIVACY_REQUEST_CALLBACK_RESUME])
+        privacy_request.status = PrivacyRequestStatus.paused
+        privacy_request.save(db)
+
+        privacy_request.cache_paused_step_and_collection(
+            ActionType.access, CollectionAddress("manual_example", "filing_cabinet")
+        )
+        response = api_client.post(url, headers=auth_header, json=[{"mock": "row"}])
+        assert response.status_code == 422
+        assert (
+            response.json()["detail"]
+            == "Cannot save manual rows. No collection in graph with name: 'manual_example:filing_cabinet'."
+        )
+
+    def test_resume_with_manual_input_invalid_data(
```

Comment:
Since `postgres_example_test_dataset_config` and `manual_dataset_config` are fixtures that aren't directly accessed:

```diff
- def test_resume_with_manual_input_invalid_data(
-    self,
-    db,
-    api_client,
-    url,
-    generate_auth_header,
-    privacy_request,
-    postgres_example_test_dataset_config,
-    manual_dataset_config,
- )
+ @pytest.mark.usefixtures("postgres_example_test_dataset_config", "manual_dataset_config")
+ def test_resume_with_manual_input_invalid_data(
+    self,
+    db,
+    api_client,
+    url,
+    generate_auth_header,
+    privacy_request,
+ )

Reply:
Thank you, I wasn't aware of `@pytest.mark.usefixtures`